bundles/org.openhab.binding.roborock/src/main/java/org/openhab/binding/roborock/internal/RoborockVacuumHandler.java

```diff
@@ -0,0 +1,1015 @@
+                return;
+            }
+            if (channelUID.getId().equals(RobotCapabilities.MOP_MODE.getChannel())) {
+                sendRPCCommand(COMMAND_SET_MOP_MODE, "[" + command.toString() + "]");
+                return;
+            }
+            if (channelUID.getId().equals(RobotCapabilities.SEGMENT_CLEAN.getChannel()) && !command.toString().isEmpty()
+                    && !command.toString().contentEquals("-")) {
+                sendRPCCommand(COMMAND_START_SEGMENT, "[" + command.toString() + "]");
+                updateState(RobotCapabilities.SEGMENT_CLEAN.getChannel(), new StringType("-"));
+                return;
+            }
+            if (channelUID.getId().equals(CHANNEL_FAN_CONTROL)) {
+                if (Integer.valueOf(command.toString()) > 0) {
+                    sendRPCCommand(COMMAND_SET_MODE, "[" + command.toString() + "]");
+                }
+                return;
+            }
+            if (channelUID.getId().equals(CHANNEL_CONSUMABLE_RESET)) {
+                sendRPCCommand(COMMAND_CONSUMABLES_RESET, "[" + command.toString() + "]");
+                updateState(CHANNEL_CONSUMABLE_RESET, new StringType("none"));
+            }
+
+            if (channelUID.getId().equals(RobotCapabilities.COLLECT_DUST.getChannel()) && !command.toString().isEmpty()
+                    && !command.toString().contentEquals("-")) {
+                sendRPCCommand(COMMAND_SET_COLLECT_DUST);
+                return;
+            }
+
+            if (channelUID.getId().equals(RobotCapabilities.CLEAN_MOP_START.getChannel())
+                    && !command.toString().isEmpty() && !command.toString().contentEquals("-")) {
+                sendRPCCommand(COMMAND_SET_CLEAN_MOP_START);
+                return;
+            }
+            if (channelUID.getId().equals(RobotCapabilities.CLEAN_MOP_STOP.getChannel())
+                    && !command.toString().isEmpty() && !command.toString().contentEquals("-")) {
+                sendRPCCommand(COMMAND_SET_CLEAN_MOP_STOP);
+                return;
+            }
+        } catch (UnsupportedEncodingException e) {
+            logger.debug("UnsupportedEncodingException, {}", e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(RoborockVacuumConfiguration.class);
+
+        if (!(getBridge() instanceof Bridge bridge
+                && bridge.getHandler() instanceof RoborockAccountHandler accountHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "@text/offline.conf-error.no-bridge");
+            return;
+        }
+        bridgeHandler = accountHandler;
+        hasChannelStructure = false;
+
+        initTask.setNamePrefix(getThing().getUID().getId());
+        pollTask.setNamePrefix(getThing().getUID().getId());
+        initTask.schedule(5);
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Prevent race conditions by first setting the state and then scheduleing the task.
```suggestion
        updateStatus(ThingStatus.UNKNOWN);
        initTask.schedule(5);
```